logstash-core/spec/logstash/output_delegator_spec.rb

```diff
@@ -15,10 +15,20 @@
 
     before do
       allow(out_klass).to receive(:new).with(any_args).and_return(out_inst)
+      allow(out_inst).to receive(:register)
+      allow(out_inst).to receive(:identifier_name)
+      allow(out_inst).to receive(:metric=).with(any_args)
+      allow(logger).to receive(:debug).with(any_args)
+    end
+
+    before do
```

Comment:
I would advise to use always ```before(:each)```  or ```before(:all)```, as relying on the default it gets me always confused for the expected behaviour.